docs/design/seccomp.md

```diff
@@ -0,0 +1,364 @@
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for adding [seccomp](https://github.com/seccomp/libseccomp) support
+to Kubernetes.  Seccomp is a system call filtering facility in the Linux
+kernel which lets applications define limits on system calls they may make,
+and what should happen when system calls are made.  Seccomp is used to reduce
+the attack surface available to applications.
+
+## Motivation
+
+Applications use seccomp to restrict the set of system calls they can make.
+Recently, container runtimes have begun adding features to allow the runtime
+to interract with seccomp on behalf of the application, which eliminates the
+need for applications to link against libseccomp directly.  Adding support in
+the Kubernetes API for describing seccomp profiles will allow administrators
+greater control over the security of workloads running in Kubernetes.
+
+Goals of this design:
+
+1.  Describe how to model seccomp profiles
+2.  Describe how to reference those profiles in the containers that use them
+
+## Constraints and Assumptions
+
+This design should:
+
+*  build upon previous security context work
+*  be container-runtime agnostic
+*  allow definition of custom profiles
+*  facilitate containerized applications that link directly to libseccomp
+
+## Use Cases
+
+1.  As an administrator, I want to be able to control access to seccomp profiles:
+    1.  To be able to default the seccomp profile when unspecified
+    2.  To be able to grant access to a seccomp profile to a class of users
+2.  As a user, I want to run an application with a seccomp profile similar to
+    the default one provided by my container runtime
+3.  As a user, I want to run an application which is already libseccomp-aware
+    in a container, and for my application to manage interacting with seccomp
+    unmediated by Kubernetes
+4.  As a user, I want to be able to describe a custom seccomp profile and use
+    it with my containers
+
+### Use Case: Administrator access control
+
+Controlling to control access to seccomp profiles is a cluster administrator concern.
+It should be possible for an administrator to:
+
+1.  Define a default seccomp profile for a class of users
+2.  Grant access to a seccomp profile to a class of users
+
+Administrators should be able to control two aspects of defaults for seccomp
+profiles:
+
+1.  Defaults for classes of users based on pod security policy
+2.  Defaults at the API level
+
+#### Classes of users
+
+The [pod security policy](https://github.com/kubernetes/kubernetes/pull/7893)
+API extension governs the ability of users to make requests that affect pod
+and container security contexts.  The proposed design should deal with
+required changes to control access to new API features added.
```

Comment:
I don't understand this paragraph. Is seccomp something that should be added to SecurityContext and PodSecurityPolicy?